cwms-data-api/src/test/java/cwms/cda/data/dao/watersupply/WaterContractDaoTestIT.java

```diff
@@ -0,0 +1,386 @@
+            contractType.add(new LookupType.Builder()
+                    .withTooltip("Test Tooltip")
+                    .withActive(true)
+                    .withDisplayValue("Test Display Value")
+                    .withOfficeId(OFFICE_ID).build());
+            contractType.add(new LookupType.Builder()
+                    .withTooltip("Test Tooltip 2")
+                    .withActive(true)
+                    .withDisplayValue("Test Display Value 2")
+                    .withOfficeId(OFFICE_ID).build());
+            dao.storeWaterContractTypes(contractType, false);
+            List<LookupType> results = dao.getAllWaterContractTypes(OFFICE_ID);
+            DTOMatch.assertMatch(contractType.get(0), results.get(0));
+            DTOMatch.assertMatch(contractType.get(1), results.get(1));
+        });
+    }
+
+    @Test
+    void testRenameWaterUser() throws Exception {
+        WaterUser newUser = buildTestWaterUser("TEST RENAME USER");
+        WaterUser updatedUser = buildTestWaterUser("TEST RENAME USER 2");
+        CwmsDatabaseContainer<?> db = CwmsDataApiSetupCallback.getDatabaseLink();
+        db.connection(c -> {
+            DSLContext ctx = getDslContext(c, OFFICE_ID);
+            WaterContractDao dao = new WaterContractDao(ctx);
+            dao.storeWaterUser(newUser, true);
+            dao.renameWaterUser(newUser.getEntityName(), updatedUser.getEntityName(), newUser.getProjectId());
+            WaterUser retrievedUser = dao.getWaterUser(newUser.getProjectId(), updatedUser.getEntityName());
+            DTOMatch.assertMatch(updatedUser, retrievedUser);
+        });
+        cleanupUserRoutine(updatedUser);
+    }
+
+    @Test
+    void testRenameWaterContract() throws Exception {
+        WaterUserContract oldContract = buildTestWaterContract("Test Old Name", true);
+        WaterUserContract renamedContract = buildTestWaterContract("Test New Name", true);
+        CwmsDatabaseContainer<?> db = CwmsDataApiSetupCallback.getDatabaseLink();
+        db.connection(c -> {
+            DSLContext ctx = getDslContext(c, OFFICE_ID);
+            WaterContractDao dao = new WaterContractDao(ctx);
+            dao.storeWaterUser(oldContract.getWaterUser(), true);
+            dao.storeWaterContract(oldContract, true, false);
+            WaterUser user = new WaterUser.Builder().withEntityName(oldContract.getWaterUser().getEntityName())
+                    .withProjectId(oldContract.getWaterUser().getProjectId())
+                    .withWaterRight(oldContract.getContractId().getName()).build();
+            dao.renameWaterContract(user, oldContract.getContractId().getName(),
+                    renamedContract.getContractId().getName());
+            List<WaterUserContract> retrievedContracts = dao.getAllWaterContracts(
+                    renamedContract.getWaterUser().getProjectId(), renamedContract.getWaterUser()
+                            .getEntityName());
+            assertFalse(retrievedContracts.isEmpty());
```

Comment:
should also test that the old contract is gone

Reply:
Added check for contract no longer in DB